lib/usb/elgato-mini.js

```diff
@@ -20,30 +20,42 @@ var debug = require('debug')('lib/usb/elgato_mini')
 var elgato_base = require('./elgato-base')
 var sharp = require('sharp')
 var image_write_queue = require('./image-write-queue')
+var setTimeoutPromise = util.promisify(setTimeout);
 
 function elgato_mini(system, devicepath) {
 	var self = this
 
 	elgato_base.apply(this, [system, devicepath, 'Elgato Streamdeck Mini'])
 
-	self.write_queue = new image_write_queue(function (key, buffer) {
-		return sharp(buffer, { raw: { width: 72, height: 72, channels: 3 } })
-			.resize(80, 80)
-			.raw()
-			.toBuffer()
-			.catch(function (e) {
-				self.system.emit('log', 'device(' + self.serialnumber + ')', 'debug', `scale image failed: ${e}`)
-				self.system.emit('elgatodm_remove_device', self.devicepath)
-			})
-			.then(function (newbuffer) {
-				if (newbuffer) {
-					self.streamDeck.fillImage(key, newbuffer)
+	self.write_queue = new image_write_queue(async function (key, buffer) {
+		let newbuffer
+		try {
+			newbuffer = await sharp(buffer, { raw: { width: 72, height: 72, channels: 3 } })
+				.resize(80, 80)
+				.raw()
+				.toBuffer()
+		} catch (e) {
+			self.system.emit('log', 'device(' + self.serialnumber + ')', 'debug', `scale image failed: ${e}`)
+			self.system.emit('elgatodm_remove_device', self.devicepath)
+			return
+		}
+
+		let attempts = 0
+		while (true) {
```

Comment:
Please make this be a for loop.
`for (let attempts=0; attempts < 3; attempts++) {`

`while(true)` is a fairly risky line of code, and much more likely to accidentally turn into an infinite loop when someone comes back here to make changes in the future.

The rest of the async changes look good :)